src/main/scala/de/upb/cs/swt/delphi/instanceregistry/daos/DatabaseInstanceDAO.scala

```diff
@@ -0,0 +1,544 @@
+package de.upb.cs.swt.delphi.instanceregistry.daos
+
+import java.io.{File, IOException, PrintWriter}
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import de.upb.cs.swt.delphi.instanceregistry.io.swagger.client.model.EventEnums.EventType
+import de.upb.cs.swt.delphi.instanceregistry.{AppLogging, Configuration, Registry}
+import de.upb.cs.swt.delphi.instanceregistry.io.swagger.client.model.InstanceEnums.{ComponentType, InstanceState}
+import de.upb.cs.swt.delphi.instanceregistry.io.swagger.client.model.LinkEnums.LinkState
+import de.upb.cs.swt.delphi.instanceregistry.io.swagger.client.model.{InstanceJsonSupport, _}
+import slick.lifted.TableQuery
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+import slick.jdbc.MySQLProfile.api._
+import slick.jdbc.meta.MTable
+import spray.json._
+
+import scala.concurrent.duration.Duration
+import scala.io.Source
+
+
+class DatabaseInstanceDAO (configuration : Configuration) extends InstanceDAO with AppLogging with InstanceJsonSupport with EventJsonSupport{
+
+  private val instances : TableQuery[Instances] = TableQuery[Instances]
+  private val instanceMatchingResults : TableQuery[InstanceMatchingResults] = TableQuery[InstanceMatchingResults]
+  private val instanceEvents : TableQuery[InstanceEvents] = TableQuery[InstanceEvents]
+  private val instanceLinks : TableQuery[InstanceLinks] = TableQuery[InstanceLinks]
+
+  implicit val system : ActorSystem = Registry.system
+  implicit val materializer : ActorMaterializer = ActorMaterializer()
+  implicit val ec : ExecutionContext = system.dispatcher
+
+  val db = Database.forURL(configuration.databaseHost + configuration.databaseName, driver = configuration.databaseDriver, user = configuration.databaseUsername, password = configuration.databasePassword)
+
+  override def addInstance(instance : Instance) : Try[Long] = {
+
+    val id = 0L //Will be set by DB
+    val host = instance.host
+    val port = instance.portNumber
+    val name = instance.name
+    val componentType = instance.componentType.toString
+    val dockerId = instance.dockerId.getOrElse("")
```

Comment:
Having an empty string in the DB is *not* the same as having the value ```None```. In your current code you are converting None to empty string, and when you read it back from the DB you are converting the empty string to ```Some("")```. This way the registry thinks that *all instances* are docker instances, even if in reality they are not. I would suggest to allow NULL in the table for instances.dockerId, so that the None value can be represented by NULL in the DB.